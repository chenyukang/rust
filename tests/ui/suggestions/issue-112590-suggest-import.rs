pub struct S;

impl fmt::Debug for S { //~ ERROR failed to resolve: use of undeclared crate or module `fmt`
    fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result { //~ ERROR failed to resolve: use of undeclared crate or module `fmt`
        //~^ ERROR failed to resolve: use of undeclared crate or module `fmt`
        Ok(())
    }
}

impl fmt::Invalid::Debug for S {} //~ ERROR failed to resolve: use of undeclared crate or module `fmt`

// will not suggest for not matching `Invalid`
impl fmt::Invalid for S {} //~ ERROR failed to resolve: use of undeclared crate or module `fmt`

type A = cell::RefCell; //~ ERROR failed to resolve: use of undeclared crate or module `cell`
type B = cell::RefError; //~ ERROR failed to resolve: use of undeclared crate or module `cell`

fn main() { }
