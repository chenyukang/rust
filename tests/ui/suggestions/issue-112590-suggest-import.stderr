error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> $DIR/issue-112590-suggest-import.rs:10:6
   |
LL | impl fmt::Invalid::Debug for S {}
   |      ^^^ use of undeclared crate or module `fmt`

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> $DIR/issue-112590-suggest-import.rs:3:6
   |
LL | impl fmt::Debug for S {
   |      ^^^ use of undeclared crate or module `fmt`
   |
help: consider importing this module
   |
LL + use std::fmt;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> $DIR/issue-112590-suggest-import.rs:4:28
   |
LL |     fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |                            ^^^ use of undeclared crate or module `fmt`
   |
help: consider importing this module
   |
LL + use std::fmt;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> $DIR/issue-112590-suggest-import.rs:4:51
   |
LL |     fn fmt(&self, _f: &mut fmt::Formatter<'_>) -> fmt::Result {
   |                                                   ^^^ use of undeclared crate or module `fmt`
   |
help: consider importing this module
   |
LL + use std::fmt;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `cell`
  --> $DIR/issue-112590-suggest-import.rs:15:10
   |
LL | type A = cell::RefCell;
   |          ^^^^ use of undeclared crate or module `cell`
   |
help: consider importing this module
   |
LL + use std::cell;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `fmt`
  --> $DIR/issue-112590-suggest-import.rs:13:6
   |
LL | impl fmt::Invalid for S {}
   |      ^^^ use of undeclared crate or module `fmt`

error[E0433]: failed to resolve: use of undeclared crate or module `cell`
  --> $DIR/issue-112590-suggest-import.rs:16:10
   |
LL | type B = cell::RefError;
   |          ^^^^ use of undeclared crate or module `cell`

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0433`.
